### PR TITLE
Audio: Component: Update to component.h module adapter conversions

### PR DESCRIPTION
**Pull request by singalsu**
In component.h the eq_iir and mixer have still the old component API function prototypes. They can be removed as not needed. These components do not support legacy mode.

Similar update is done to testbench common_test.h. Remove eq_iir and mixer old init function prototypes and add to module adapter style init functions mfcc and mixer those were missing.

### CODE REVIEW

**Review by kv2019i — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
SOFCI TEST

File: tools/testbench/include/testbench/common_test.h [hunk 1]
```diff
@@ -118,6 +116,8 @@ void sys_comp_tdfb_init(void);
 void sys_comp_module_demux_interface_init(void);
 void sys_comp_module_eq_fir_interface_init(void);
 void sys_comp_module_eq_iir_interface_init(void);
+void sys_comp_module_mfcc_interface_init(void);
+void sys_comp_module_mixer_interface_init(void);
```
Comment (lyakh) [inline tools/testbench/include/testbench/common_test.h:120]: if it worked until now without these - does it mean, that this isn't compile-tested by the CI or maybe that these prototypes aren't needed and those functions can be static?

Comment (singalsu) [reply tools/testbench/include/testbench/common_test.h:120]: I wonder if recent testbench patches have forgotten these? But these will definitely be needed to load the components (unless they get deprecate - e.g. mixer in favor of mixin/mixout?) and run them.

Comment (singalsu) [reply tools/testbench/include/testbench/common_test.h:120]: @lyakh Sorry, I was wrong. These protypes are no more necessary. The component.h is used, need to delete all these in the patch.

**Review by lyakh — APPROVED** [review]

**Comment by kv2019i** [comment]
SOFCI TEST

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**